keyvault/client/src/main/java/com/azure/keyvault/SecretClient.java

```diff
@@ -0,0 +1,540 @@
+        return service.setSecret(vaultEndPoint, name, API_VERSION, ACCEPT_LANGUAGE, parameters);
+    }
+
+
+    /**
+     * Get a specified secret from the key vault.
+     * The GET operation is applicable to any secret stored in Azure Key Vault.
+     * This operation requires the secrets/get permission.
+     *
+     * @param name The name of the secret, cannot be null
+     * @param version The version of the secret, cannot be null
+     * @throws NullPointerException if {@code name} or {@code version} parameter is {@code null}.
+     * @return the Secret that was requested.
+     */
+    public Mono<RestResponse<Secret>> getSecretAsync(String name, String version) {
+        Objects.requireNonNull(name, "The Secret name cannot be null.");
+        Objects.requireNonNull(version, "The Secret version cannot be null.");
+        return service.getSecret(vaultEndPoint, name, version, API_VERSION, ACCEPT_LANGUAGE);
+    }
+
+    /**
+     * Get a specified secret from the key vault.
+     * The GET operation is applicable to any secret stored in Azure Key Vault.
+     * This operation requires the secrets/get permission.
+     *
+     * @param name The name of the secret.
+     * @throws NullPointerException if {@code name} parameter is {@code null}.
+     * @return the Secret that was requested.
+     */
+    public Mono<RestResponse<Secret>> getSecretAsync(String name) {
+        Objects.requireNonNull(name, "The Secret name cannot be null.");
+        return getSecretAsync(name, "");
+    }
+
+
+    /**
+     * Updates the attributes associated with a specified secret in the key vault.
+     * The UPDATE operation changes specified attributes of an existing stored secret.
+     * Attributes that are not specified in the request are left unchanged.
+     * The value of a secret itself cannot be changed. This operation requires the secrets/set permission.
+     *
+     * @param secret the Secret object with updated properties.
+     * @throws NullPointerException if {@code secret} is {@code null}.
+     * @return the Secret that was updated.
+     */
+    public Mono<RestResponse<Secret>> updateSecretAsync(Secret secret) {
+        Objects.requireNonNull(secret, "The Secret input parameter cannot be null.");
+        Objects.requireNonNull(secret.name(), "The Secret name cannot be null.");
+
+        SecretRequestAttributes secretRequestAttributes = new SecretRequestAttributes()
+                .withEnabled(secret.enabled())
+                .withExpires(secret.expires())
+                .withNotBefore(secret.notBefore());
+
+        SecretRequestOptions parameters = new SecretRequestOptions()
+                .withValue(secret.value())
+                .withTags(secret.tags())
+                .withContentType(secret.contentType())
+                .withSecretAttributes(secretRequestAttributes);
+
+        return service.updateSecret(vaultEndPoint, secret.name(), "", API_VERSION, ACCEPT_LANGUAGE, parameters);
```

Comment:
>return service.updateSecret(vaultEndPoint, secret.name(), "", API_VERSION, ACCEPT_LANGUAGE, parameters); [](start = 8, length = 104)

Need to specify the secret.version() rather than ""

Reply:
updated.